### PR TITLE
[ENG-2334] Make DOIs mint on project/draft registration 

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

This is primarily a ticket for ensuring that DOIs mint of project draft registration, but at some time in the distant past the ability to mint DOIs locally and therefore test this feature was disabled, so I had to make some tweaks in order to restore that behavior.

## Changes

- tweaks client (apparently just flat broken) to allow for minting DOIs and restructure tests to be mocked properly.
- removes old ServiceTemporarilyUnavailable exception
- removes old settings checks that were getting in the way of tests.  

## QA Notes

- Verify at https://doi.test.datacite.org/repositories/demo.cos/dois that DOIs are being minted when a project or draft reg is registered. ⚠️ You will need appropriate credentials to do this. ⚠️ 
- Verify Registration process is timely and error free.

## Documentation

QA: is there any user facing docs about this?

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2334

### CODE REVIEW

**Comment by corbinSanders** [comment]
LGTM!

File: osf/models/node.py [hunk 1]
```diff
@@ -2321,10 +2321,7 @@ def file_write_scope(self):
         return oauth_scopes.CoreScopes.NODE_FILE_WRITE
 
     def get_doi_client(self):
-        if settings.DATACITE_URL and settings.DATACITE_PREFIX:
-            return DataCiteClient(base_url=settings.DATACITE_URL, prefix=settings.DATACITE_PREFIX)
-        else:
-            return None
+        return DataCiteClient(base_url=settings.DATACITE_URL, prefix=settings.DATACITE_PREFIX)
```
Comment (brianjgeiger) [inline osf/models/node.py:2324]: What happens if the DATACITE_URL or DATACITE_PREFIX aren't set? Does it result in an exception that we would catch? Or do we always have _something_ for these settings now?

**Review by brianjgeiger — COMMENTED** [review]
Looking good. One question.

File: osf/models/node.py (hunk 1 above)
Comment (Johnetordoff) [reply osf/models/node.py:2324]: They should always be set now, just different for access prod vs. staging.

**Review by brianjgeiger — APPROVED** [review]

**Pull request merged**